### PR TITLE
codeql: 2.5.9 -> 2.6.2, support darwin

### PR DESCRIPTION
**Pull request by hmac**
###### Motivation for this change
This change updates CodeQL to the latest version, and makes some changes so it builds on macOS.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 136592` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/development/tools/analysis/codeql/default.nix [hunk 1]
```diff
@@ -1,26 +1,18 @@
...
-, zlib
-, xorg
-, freetype
-, alsa-lib
-, jdk11
-, curl
-, lttng-ust
-, autoPatchelfHook
-}:
+{ lib, stdenv, fetchzip, zlib, xorg, freetype, alsa-lib, jdk11, curl, lttng-ust
```
Comment (austinbutler) [inline pkgs/development/tools/analysis/codeql/default.nix:1]: Remove unused inputs:

```suggestion
{ lib, stdenv, fetchzip, zlib, xorg, freetype, jdk11, curl
```

File: pkgs/development/tools/analysis/codeql/default.nix [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
+, autoPatchelfHook }:
 
 stdenv.mkDerivation rec {
   pname = "codeql";
-  version = "2.5.9";
+  version = "2.6.0";
```
Comment (austinbutler) [inline pkgs/development/tools/analysis/codeql/default.nix:6]: 2.6.2 is out now FYI.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 136592` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>codeql</li>
  </ul>
</details>

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 136592` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>codeql</li>
  </ul>
</details>

**Review by austinbutler — CHANGES_REQUESTED** [review]

**Comment by austinbutler** [comment]
Seems to build and run on Linux and Darwinl

File: pkgs/development/tools/analysis/codeql/default.nix [hunk 3]
```diff
@@ -1,26 +1,18 @@
...
 
   dontConfigure = true;
   dontBuild = true;
   dontStrip = true;
 
   src = fetchzip {
-    url = "https://github.com/github/codeql-cli-binaries/releases/download/v${version}/codeql.zip";
-    sha256 = "sha256-r3Jm+VYjn0Dz4BCSbADbgTWL1owbyIXlkoj6mOmZcZk=";
+    url =
+      "https://github.com/github/codeql-cli-binaries/releases/download/v${version}/codeql.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/codeql/default.nix:14]:
```suggestion
    url = "https://github.com/github/codeql-cli-binaries/releases/download/v${version}/codeql.zip";
```

File: pkgs/development/tools/analysis/codeql/default.nix (hunk 2 above)
Comment (hmac) [reply pkgs/development/tools/analysis/codeql/default.nix]: I've bumped this to 2.6.2.

File: pkgs/development/tools/analysis/codeql/default.nix (hunk 3 above)
Comment (hmac) [reply pkgs/development/tools/analysis/codeql/default.nix:13]: I've run this file through both `nixfmt` and `nixpkgs-fmt` and both produce this output. Are you sure we want to change it?

Comment (SuperSandro2000) [reply pkgs/development/tools/analysis/codeql/default.nix:13]: Yes, I am sure. nixfmt has an old fashioned idea about line lengths and nixpkgs-fmt does not reverse that decission.

**Review by jollheef — APPROVED** [review]
LGTM

**Review by austinbutler — APPROVED** [review]
Looks good, although you probably want to squash the commits down to what's relevant now. For example, don't need "codeql: 2.5.9 -> 2.6.0".

**Pull request merged**